logic/src/commonMain/kotlin/com/wire/kalium/logic/data/e2ei/CertificateRevocationListRepository.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.data.e2ei
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.configuration.UserConfigRepository
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.map
+import com.wire.kalium.logic.wrapApiRequest
+import com.wire.kalium.network.api.base.unbound.acme.ACMEApi
+import com.wire.kalium.persistence.config.CRLUrlExpirationList
+import com.wire.kalium.persistence.config.CRLWithExpiration
+import com.wire.kalium.persistence.dao.MetadataDAO
+import io.ktor.http.Url
+import io.ktor.http.protocolWithAuthority
+
+interface CertificateRevocationListRepository {
+
+    /**
+     * Returns CRLs with expiration time.
+     *
+     * @return the [CRLUrlExpirationList] representing a list of CRLs with expiration time.
+     */
+    suspend fun getCRLs(): CRLUrlExpirationList?
+    suspend fun addOrUpdateCRL(url: String, timestamp: ULong)
+    suspend fun getCurrentClientCrlUrl(): Either<CoreFailure, String>
+    suspend fun getClientDomainCRL(url: String): Either<CoreFailure, ByteArray>
+}
+
+internal class CertificateRevocationListRepositoryDataSource(
+    private val acmeApi: ACMEApi,
+    private val metadataDAO: MetadataDAO,
+    private val userConfigRepository: UserConfigRepository
+) : CertificateRevocationListRepository {
+    override suspend fun getCRLs(): CRLUrlExpirationList? =
+        metadataDAO.getSerializable(CRL_LIST_KEY, CRLUrlExpirationList.serializer())
+
+    override suspend fun addOrUpdateCRL(url: String, timestamp: ULong) {
+
+        metadataDAO.getSerializable(CRL_LIST_KEY, CRLUrlExpirationList.serializer())
```

Comment:
hmm, what if there is no any data for the `CRL_LIST_KEY`  yet?  
It will return `null` and we don't add any CRL (because of `?.let{}` ). Maybe we should create an empty list, add a new item to it and putSerializable?

Reply:
There is `?: run { }` that will handle that case and add it

Reply:
@borichellow You are right, pushing the fix now..